PixivHelper.py

```diff
@@ -553,18 +553,21 @@ def have_strings(page, strings):
 
 
 def get_ids_from_csv(ids_str, sep=',', is_string=False):
-    ids = list()
-    ids_str = str(ids_str).split(sep)
-    for id_str in ids_str:
-        temp = id_str.strip()
-        if len(temp) > 0:
-            try:
-                _id = temp
-                if not is_string:
-                    _id = int(temp)
-                ids.append(_id)
-            except ValueError:
-                print_and_log('error', u"ID: {0} is not valid".format(id_str))
+    ids = []
+    if is_string:
+        ids_str = str(ids_str).split(sep)
+        for id_str in ids_str:
+            temp = id_str.strip()
+            if len(temp) > 0:
+                try:
+                    _id = temp
+                    ids.append(_id)
+                except ValueError:
+                    print_and_log('error', u"ID: {0} is not valid".format(id_str))
+    else:
+        ids = re.findall("(?:series|users|\s|,|^|artworks)\/?(\d+)", ids_str)
+        if not ids:
+            print_and_log('error', u"ID: {0} is not valid".format(ids_str))
```

Comment:
need to change the logic because is_string is not always set to True.
![image](https://user-images.githubusercontent.com/836867/99343849-ab281380-28c9-11eb-8606-aec925334747.png)

I think you better add a check if the ids_str is start with http or not, e.g. :
``` python
if ids_str.startswith("http"):
    # check by regex
else:
    # original logic

```

Also need to take care if it is mixed case (url with normal id)?

Reply:
> need to change the logic because is_string is not always set to True.
> ![image](https://user-images.githubusercontent.com/836867/99343849-ab281380-28c9-11eb-8606-aec925334747.png)

Ah yes, it's meant for the case when `is_string=False` (so the case when we don't paste the fanbox/sketch member id/name, which would be something for the if `is_string=True` case). Thinking of it, the `is_string=True` case is also very easy to handle with urls, so I am going to add that I guess.

> I think you better add a check if the ids_str is start with http or not, e.g. :
> Also need to take care if it is mixed case (url with normal id)?

It does already that with the regex 
`ids = re.findall("(?:series|users|\s|,|^|artworks|posts)\/?(\d+)", ids_str)`
checking for numbers which are prepended by certain stuff, so we don't accidentally catch stuff like the page number or artist id when downloading series